packages/dashboards/tests/unit/services/dashboard-data-test.js

```diff
@@ -137,14 +153,85 @@ module('Unit | Service | dashboard data', function(hooks) {
       return resolve({});
     });
 
-    await service.fetchDataForWidgets(
+    service.fetchDataForWidgets(
       1,
       [{ id: 2, dashboard: cloneDeep(dashboard), requests: [makeRequest(4)] }],
+      [{ widgetId: 2 }],
       [],
       optionsObject
     );
   });
 
+  test('fetch concurrency and restart', async function(assert) {
+    assert.expect(3);
+
+    const done = assert.async();
+
+    const fetchCalls = [];
+
+    const service = this.owner.factoryFor('service:dashboard-data').create({
+      async _fetch(request) {
+        fetchCalls.push(request.data);
+        await timeout(100);
+
+        // cancel all widget tasks
+        run(() => {
+          service._fetchRequestsForWidget.cancelAll();
+        });
+
+        later(() =>
```

Comment:
Was this supposed to be `return resolve({...` ?

This test feels a little confusing with the `timeout`/`cancelAll` being run in the `_fetch`. It might be simpler to do something like this
```js
return new Promise((resolve) => {
   const resolveRequest = () => resolve({ request, response: { data: request.data } });
   resolveRequests[request.data] = resolveRequest;
});
```
So that we can manually resolve each query or not and manually call `service._fetchRequestsForWidget.cancelAll();`

Reply:
since now we won't be returning the `Task` from the service anymore and not calling `.cancelAll()`, I'll test only concurrency config here.

Reply:
actually will check that in the previous test